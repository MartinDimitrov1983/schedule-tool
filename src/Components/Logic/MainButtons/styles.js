const styles = {
  mainButtons: {
    display: 'grid',
    gridTemplateColumns: '57% 1fr 1fr 1fr',
    gap: '10px',
    justifyContent: 'end',
    margin: '20px 48px',
  },
  reset: { maxWidth: '50px' },
};

export default styles;
