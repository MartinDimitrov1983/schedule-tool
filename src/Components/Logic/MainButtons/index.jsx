import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Box } from '@mui/material';
import { Button } from '../../';
import { shouldDisableButtons, shouldDisableUpload } from '../../../Utils';
import {
  MAGENTA,
  AUTOCOMPLETE,
  RESET,
  LIGHT_DARK,
  UPLOAD,
} from '../../../Constants';
import styles from './styles';

const MainButtons = ({
  resetHandler,
  mouseEnterhandler,
  mouseLeaveHadler,
  autoCompleteHandler,
  uploadHandler,
  isAutoCompletePressed,
}) => {
  const times = useSelector((state) => state.days.times);
  return (
    <Box sx={styles.mainButtons}>
      <Box />
      <Button
        text={RESET}
        color={LIGHT_DARK}
        disabled={shouldDisableButtons(times)}
        sx={styles.reset}
        onClick={resetHandler}
      />
      <Button
        text={AUTOCOMPLETE}
        color={MAGENTA}
        disabled={shouldDisableButtons(times) || isAutoCompletePressed}
        onMouseEnter={mouseEnterhandler}
        onMouseLeave={mouseLeaveHadler}
        onClick={autoCompleteHandler}
      />
      <Button
        text={UPLOAD}
        disabled={shouldDisableUpload(times)}
        color={LIGHT_DARK}
        onClick={uploadHandler}
      />
    </Box>
  );
};

MainButtons.propTypes = {
  resetHandler: PropTypes.func,
  mouseEnterhandler: PropTypes.func,
  mouseLeaveHadler: PropTypes.func,
  autoCompleteHandler: PropTypes.func,
  uploadHandler: PropTypes.func,
  isAutoCompletePressed: PropTypes.bool,
};

export default MainButtons;
