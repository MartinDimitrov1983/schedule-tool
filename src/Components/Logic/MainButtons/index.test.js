import React from 'react';
import { render } from '@testing-library/react';
import * as Redux from 'react-redux';

import MainButtons from './index';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

test('Snapshot MainButtons', () => {
  jest.spyOn(Redux, 'useSelector').mockReturnValue([['09:00', '12:00']]);
  const { asFragment } = render(
    <MainButtons
      resetHandler={() => {}}
      mouseEnterhandler={() => {}}
      mouseLeaveHadler={() => {}}
      autoCompleteHandler={() => {}}
      uploadHandler={() => {}}
      isAutoCompletePressed={false}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
