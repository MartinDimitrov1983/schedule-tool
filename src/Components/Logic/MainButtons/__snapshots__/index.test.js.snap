// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot MainButtons 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-wdunpr"
  >
    <div
      class="MuiBox-root css-0"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-y8rsyv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Reset
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1m39mvv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Autocomplete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-y8rsyv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Upload
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
